### ISSUE TITLE
Node-resolution or glob support for packages names?

### ISSUE THREAD

**Issue by chikamichi** [systemjs/systemjs#1789]
Hi,

I'm facing an issue I'm not sure it's *actually* an issue, or just me failing :)

Here's my use-case, using JSPM:

![hierarchy](https://user-images.githubusercontent.com/106689/35009815-1b649924-fb01-11e7-9836-c48095f878d4.png)

I've got a bunch of folders at ./src/components/ which are organized with an index.ts "main" file. I want to be able to write plain paths in import statements and automatically resolve to those index.ts files.

After fiddling with `packages`, `meta`, `map` and the likes for a couple hours, including the obvious "app/components/*" (and /**/* as a variant), no luck. I went for the hard-coded version:

``` js
  // "app" refers to src/
  packages: {
    "app/": {
      "defaultExtension": "ts"
    },
    "app/components/Core": {
      "defaultExtension": "ts",
      "main": "index.ts"
    },
    "app/components/CustomText": {
      "defaultExtension": "ts",
      "main": "index.ts"
    },
    "app/components/LiveText": {
      "defaultExtension": "ts",
      "main": "index.ts"
    }
  }
```

but that's some boilerplate right here :)

As far as I understand, and as revealed through testing, `moduleResolution: "node"` won't help for it only works with node_modules/ and jspm_packages/.

Is there a way I can either replicate this behaviour with my project's path /src, or tweak `moduleResolution` into considering that path? If not, can wildcards be used in packages name somehow?

Basically, if anyone would be so kind as to point me in the right direction, much appreciated!

Thank you.

*Code is available at https://github.com/chikamichi/typometer*
